### PR TITLE
Get Block Children

### PR DESCRIPTION
**Pull request by bspradling**
## Description 

Generic implementation of retrieving block information. I have yet to enumerate different fields for specific Block types. 

Fixed some visibility issues for people using this client externally. 
Fixed an casing issue when deserializing Url types

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -133,12 +133,69 @@ pub struct Page {
...
+    ToDo,
+    Toggle,
+    ChildPage,
+    Unsupported
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct Block {
+    id: BlockId,
+    r#type: BlockType,
```
Comment (jakeswenson) [inline src/models.rs:186]: Neat, but instead what about 
```suggestion
    #[serde(rename = "type")]
    block_type: BlockType,
```

Comment (jakeswenson) [reply src/models.rs:186]: Should block actually be an `enum`?
```rust
pub enum Block {
    Paragraph {
       #[serde(flatten)]
       common: BlockCommon
    }
    ...
```

**Review by jakeswenson — CHANGES_REQUESTED** [review]

File: src/models.rs (hunk 1 above)
Comment (bspradling) [reply src/models.rs:186]: Yeah it needs to be. ill refactor it

**Comment by bspradling** [comment]
Closing in favor of #9

**Pull request closed without merging**